arangod/IResearch/IResearchViewCoordinator.cpp

```diff
@@ -221,11 +223,10 @@ arangodb::Result IResearchViewCoordinator::appendVelocyPackImpl(
   if (!hasFlag(flags, Serialize::ForPersistence)) {
     // verify that the current user has access on all linked collections
     ExecContext const& exec = ExecContext::current();
-    if (!exec.isSuperuser()) {
```

Comment:
If the user is a superuser then the user should have access to the collection. Therefore removing this short-cut as Superuser is not well-defined.